test/byContract/registries/resources.test.ts

```diff
@@ -0,0 +1,281 @@
+            await iexecPoco.datasetregistry(),
+            anyone,
+        );
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            await iexecPoco.workerpoolregistry(),
+            anyone,
+        );
+
+        ensRegistry = ENSRegistry__factory.connect(ensRegistryAddress, anyone);
+        const reverseRootNameHash = ethers.utils.namehash('addr.reverse');
+        const reverseRegistrarAddress = await ensRegistry.owner(reverseRootNameHash);
+        const reverseResolverAddress = await ReverseRegistrar__factory.connect(
+            reverseRegistrarAddress,
+            anyone,
+        ).defaultResolver();
+        reverseResolver = PublicResolver__factory.connect(reverseResolverAddress, anyone);
+    }
+
+    describe('App', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            MULTIADDR_BYTES,
+            ethers.utils.id(`Content of my app`),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+        beforeEach(async () => {
+            const appAddress = await appRegistry.callStatic.createApp(
+                appProvider.address,
+                ...createAppArgs,
+            );
+            await appRegistry
+                .createApp(appProvider.address, ...createAppArgs)
+                .then((tx) => tx.wait());
+            app = App__factory.connect(appAddress, anyone);
+        });
+
+        describe('creation and initialization', () => {
+            it('Should create an app and verify its details', async () => {
+                expect(await app.registry()).to.equal(appRegistry.address);
+                expect(await app.owner()).to.equal(appProvider.address);
+                expect(await app.m_appName()).to.equal(createAppArgs[0]);
+                expect(await app.m_appType()).to.equal(createAppArgs[1]);
+                expect(await app.m_appMultiaddr()).to.equal(createAppArgs[2]);
+                expect(await app.m_appChecksum()).to.equal(createAppArgs[3]);
+                expect(await app.m_appMREnclave()).to.equal(createAppArgs[4]);
+            });
+
+            it('Should not reinitialize created app', async () => {
+                await expect(app.initialize(...createAppArgs)).to.be.revertedWith(
+                    'already initialized',
+                );
+            });
+        });
+
+        describe('setName', () => {
+            it('Should set name for the ENS reverse registration of the app', async () => {
+                const newENSName = 'myApp.eth';
+                await app
+                    .connect(appProvider)
+                    .setName(ensRegistry.address, newENSName)
+                    .then((tx) => tx.wait());
+                expect(await reverseResolver.name(computeNameHash(app.address))).to.equal(
+                    newENSName,
+                );
+            });
+
+            it('Should revert when a non-owner tries to set the ENS name', async () => {
```

Comment:
```suggestion
            it('Should not set name when sender is not the owner', async () => {
```
to be coherent with other tests.

Reply:
Updated [here](https://github.com/iExecBlockchainComputing/PoCo/pull/125/commits/6609d74da52f19de2bfdd8be46ac1e236e4a0e5f)